gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -402,9 +408,9 @@ void SDLAudio::buffer_callback(void *udata, uint8_t *stream, int len)
 	int mixerLen = len;
 
 	SDLAudio *driver = (SDLAudio *)udata;
-	std::lock_guard<std::recursive_mutex> l(driver->OurMutex);
-
 	unsigned int remaining = len;
+	driver->MusicMutex.lock();
```

Comment:
please use a `lock_guard` instead of manually locking/unlocking so that changes to the code behave expectedly. there is just a lot of code between the 2.

Reply:
But there's no need to lock anything at `SetAudioStreamVolume` anymore. Or even at `memset` before that. Is it still better to use `lock_guard` in that case?

Reply:
this is another one of those infinite loops where a `lock_guard` should be at the top of the loop instead of outside the loop.